draft-bormann-core-corr-clar.md

```diff
@@ -488,6 +488,69 @@ issue include:
+In those situations, a server still needs to maintain its security and and amplification mitigation properties,
+which are generally independent topics but can be addressed using the same tools.
+
+A safe option is always to reject the initial request and request confirmation.
+The RECOMMENDED way to do that is using CoAP's mechanism of sending a 4.01 (Unauthorized) response with an Echo option
+(where a subsequent request with the same Echo value proves to the server that the destination was reachable);
+clients SHOULD act to the Echo option as specified in {{?RFC9175}}.
+Tools specific to a security mechanism such as RRC ({{?I-D.ietf-tls-dtls-rrc}} may be used instead,
+but their use may depend on successful negotiation.
+
+### Amplification mitigation and return routability
+
+If it is not certain that the client is reachable on the request's sender address,
+but the response does not exceed the request's size by a factor of 3 ({{Section 2.4 of RFC9175}}, item 3),
+the server can answer the request.
+It should still include an Echo value, whose presence in the next request serves to confirm the client's address.
+
+This situation can happen at any time in OSCORE,
+or in DTLS after a CID based resumption.
+
+Verifying the client's address is not only relevant for amplification attacks
+(which addresses attacks described in {{?I-D.irtf-t2trg-amplification-attacks}})
+but also for traffic misdirection.
+{{Section 7 of ?I-D.ietf-tls-dtls-rrc}} contains a menu of options how to use RRC messages to distinguish different cases.
+An 4.01 response with Echo can perform some of the functions equivalently
+(with the Echo value taking the place of the RRC cookie),
+but does not provide a means to distinguish between non-preferred and preferred paths.
+Where that distinction matters,
+RRC provides the right tools to make it.
+
+### Replay protection
+
+If it is not certain that the request is not a replay,
+but the request handler is safe or long-term idempotent
+and there is no risk of metadata revealing data,
+the server can answer the request.
+Metadata that can reveal data are the size of the response
+(which, in a replay situation, can give an active attacker additional data)
+as well as any processing delays.
+(There should be no observable side effects for safe or previously processed idempotent requests).
+
+Assessing whether a resource is long-term idempotent is not always trivial, and it is prudent to err at the side of caution.
+If nothing else, GET requests to constant resources,
+such as queries to /.well-known/core,
+can often be responded to safely on the CoAP layer even without any replay protection.
+
+This situation can happen in OSCORE after a partial loss of context.
+It can currently not happen in DTLS because 0-RTT Data is not allowed for CoAP (cf. {{Section 14 of ?I-D.ietf-uta-tls13-iot-profile-09}})
+at the time of writing
+(but that may change with a future document).
```

Comment:
```suggestion
Currently, this cannot happen in DTLS because 0-RTT Data is not allowed for CoAP (cf. {{Section 14 of ?I-D.ietf-uta-tls13-iot-profile-09}}). However, that may change if a future document defines a profile for using early data with CoAP.
```